src/core/simulation_sequence.jl

```diff
@@ -132,50 +133,57 @@ end
 mutable struct SimulationSequence
     horizons::Dict{String, Int}
     step_resolution::Dates.TimePeriod
-    intervals::Dict{String, <:Dates.TimePeriod}
+    intervals::Dict{String, Tuple{<:Dates.TimePeriod, <:FeedForwardChronology}}
     order::Dict{Int, String}
     feedforward_chronologies::Dict{Pair{String, String}, <:FeedForwardChronology}
     feedforward::Dict{Tuple{String, Symbol, Symbol}, <:AbstractAffectFeedForward}
     ini_cond_chronology::IniCondChronology
     cache::Dict{String, Vector{<:AbstractCache}}
     execution_order::Vector{Int}
+    current_execution_index::Int64
 
     function SimulationSequence(;
         horizons::Dict{String, Int},
         step_resolution::Dates.TimePeriod,
-        intervals::Dict{String, <:Dates.TimePeriod},
+        intervals::Dict{String, <:Tuple{<:Dates.TimePeriod, <:FeedForwardChronology}},
         order::Dict{Int, String},
         feedforward_chronologies = Dict{Pair{String, String}, FeedForwardChronology}(),
         feedforward = Dict{Tuple{String, Symbol, Symbol}, AbstractAffectFeedForward}(),
         ini_cond_chronology = InterStageChronology(),
         cache = Dict{String, Vector{AbstractCache}}(),
     )
         _check_stage_order(order)
+        _intervals = Dict{String, Tuple{<:Dates.TimePeriod, <:FeedForwardChronology}}()
+        for (k, v) in intervals
+            _intervals[k] = (IS.time_period_conversion(intervals[k][1]), intervals[k][2])
+        end
+        step_resolution = IS.time_period_conversion(step_resolution)
```

Comment:
Given the new, more explicit, definition of the sequence and order, do we still need the `step_resolution` argument? Can it be inferred from the interval of the first stage?

Reply:
No, we can't get rid of step resolution if the first stage is supposed to run more than once per step.